homeassistant/components/homely/homely_device.py

```diff
@@ -0,0 +1,132 @@
+"""Classes to contain the entity device link provided by Homely."""
+from datetime import timedelta
+import logging
+
+from requests import ConnectTimeout, HTTPError
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+from homelypy.devices import Device, SingleLocation, State
+from homelypy.homely import ConnectionFailedException, Homely
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class HomelyHome:
+    """A Homely location."""
+
+    def __init__(self, hass: HomeAssistant, entry: ConfigEntry) -> None:
+        """Set up data properties."""
+        self.hass = hass
+        self.username = entry.data["username"]
+        self.password = entry.data["password"]
+        self.location_id = entry.data["location_id"]
+        self.location: SingleLocation = None
+        self.homely: Homely = None
+        self.devices: dict[str, HomelyDevice] = {}
+        self.coordinator: DataUpdateCoordinator
+
+    async def setup(self) -> None:
+        """Perform initial setup."""
+        try:
+            self.homely = Homely(self.username, self.password)
```

Comment:
Can this raise an exception?  If not please move it outside the try block

Reply:
No, so moved outside the try block.